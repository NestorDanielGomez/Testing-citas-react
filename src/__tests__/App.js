import React from "react";
import { fireEvent, getByText, render, screen } from "@testing-library/react";
import App from "../App";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";

test("<App/>La aplicacion funciona correctamente", () => {
  render(<App />);
  expect(screen.getByText("Administrador de Mascotas")).toBeInTheDocument();
  expect(screen.getByTestId("nombre_app").textContent).toBe("Administrador de Mascotas");

  expect(screen.getByText("Crear Cita")).toBeInTheDocument();
  expect(screen.getByText("Administrador de Mascotas")).toBeInTheDocument();
});

test("<App/>Agrega cita y verifica heading", () => {
  render(<App />);
  userEvent.type(screen.getByTestId("mascota"), "Yaco");
  userEvent.type(screen.getByTestId("propietario"), "Nestor");
  userEvent.type(screen.getByTestId("fecha"), "2023-03-03");
  userEvent.type(screen.getByTestId("hora"), "10:30");
  userEvent.type(screen.getByTestId("sintomas"), "come mucho");

  const btnSubmit = screen.getByTestId("btn-submit");
  userEvent.click(btnSubmit);

  const alerta = screen.queryByTestId("alerta");
  expect(alerta).not.toBeInTheDocument();

  expect(screen.getByTestId("titulo_dinamico").textContent).toBe("Administra tus Citas");
  expect(screen.getByTestId("titulo_dinamico").textContent).not.toBe("No hay Citas");
});

test("<App/>Agrega cita en el DOM", async () => {
  render(<App />);
  const citas = await screen.queryAllByTestId("cita");
  // console.log(citas.toString());
  // expect(citas).toMatchSnapshot();
  expect(screen.getByTestId("btn_eliminar").tagName).toBe("BUTTON");
  expect(screen.getByTestId("btn_eliminar")).toBeInTheDocument();

  expect(screen.getByText("Yaco")).toBeInTheDocument();
});

test("<App/>Elimina la cita", async () => {
  render(<App />);

  // console.log(citas.toString());
  // expect(citas).toMatchSnapshot();
  expect(screen.getByTestId("btn_eliminar").tagName).toBe("BUTTON");
  expect(screen.getByTestId("btn_eliminar")).toBeInTheDocument();

  //simulo el click
  userEvent.click(screen.getByTestId("btn_eliminar"));
  //el btn ya no va a estar

  expect(screen.queryByTestId("btn_eliminar")).not.toBeInTheDocument();
  expect(screen.queryByText("Yaco")).not.toBeInTheDocument();
  expect(screen.queryByText("cita")).not.toBeInTheDocument();
});
