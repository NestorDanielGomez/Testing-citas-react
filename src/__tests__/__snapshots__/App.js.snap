// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/>Agrega cita en el DOM 1`] = `
Array [
  <div
    class="cita"
    data-testid="cita"
  >
    <p>
      Mascota: 
      <span>
        Yaco
      </span>
       
    </p>
    <p>
      Dueño: 
      <span>
        Nestor
      </span>
       
    </p>
    <p>
      Fecha: 
      <span>
        2023-03-03
      </span>
       
    </p>
    <p>
      Hora: 
      <span>
        10:30
      </span>
       
    </p>
    <p>
      Sintomas: 
      <span>
        come mucho
      </span>
       
    </p>
    <button
      class="button eliminar u-full-width"
    >
      Eliminar ×
    </button>
  </div>,
]
`;
